operators/infinispan/2.4.0/manifests/infinispan-operator.clusterserviceversion.yaml

```diff
@@ -0,0 +1,939 @@
+          - infinispan.org
+          resources:
+          - infinispans
+          - infinispans/finalizers
+          - infinispans/status
+          verbs:
+          - create
+          - get
+          - list
+          - patch
+          - update
+          - watch
+        - apiGroups:
+          - infinispan.org
+          resources:
+          - restores
+          - restores/finalizers
+          - restores/status
+          verbs:
+          - create
+          - get
+          - list
+          - patch
+          - update
+          - watch
+        - apiGroups:
+          - integreatly.org
+          resources:
+          - grafanadashboards
+          verbs:
+          - create
+          - delete
+          - get
+          - list
+          - update
+          - watch
+        - apiGroups:
+          - monitoring.coreos.com
+          resources:
+          - servicemonitors
+          verbs:
+          - create
+          - delete
+          - get
+          - list
+          - update
+          - watch
+        - apiGroups:
+          - networking.k8s.io
+          resources:
+          - customresourcedefinitions
+          - customresourcedefinitions/status
+          verbs:
+          - get
+          - list
+        - apiGroups:
+          - networking.k8s.io
+          resources:
+          - ingresses
+          verbs:
+          - create
+          - delete
+          - deletecollection
+          - get
+          - list
+          - update
+          - watch
+        - apiGroups:
+          - rbac.authorization.k8s.io
+          resources:
+          - rolebindings
+          - roles
+          verbs:
+          - create
+          - delete
+          - get
+          - list
+          - update
+          - watch
+        - apiGroups:
+          - route.openshift.io
+          resources:
+          - routes
+          - routes/custom-host
+          verbs:
+          - create
+          - delete
+          - deletecollection
+          - get
+          - list
+          - update
+          - watch
+        serviceAccountName: infinispan-operator-controller-manager
+    strategy: deployment
+  installModes:
+  - supported: true
+    type: OwnNamespace
+  - supported: true
+    type: SingleNamespace
+  - supported: true
+    type: MultiNamespace
+  - supported: true
+    type: AllNamespaces
+  keywords:
+  - infinispan
+  - key value
+  - database
+  - datagrid
+  - open source
+  links:
+  - name: Blog
+    url: https://blog.infinispan.org/
+  - name: Documentation
+    url: https://infinispan.org/documentation
+  - name: Chat
+    url: https://infinispan.zulipchat.com/#narrow/stream/185835-infinispan-cloud
+  - name: Operator Source Code
+    url: https://github.com/infinispan/infinispan-operator
+  maintainers:
+  - email: infinispan-dev@lists.jboss.org
+    name: Infinispan Community
+  maturity: alpha
+  minKubeVersion: 1.24.0
+  provider:
+    name: Infinispan
+  replaces: infinispan-operator.v2.3.7
```

Comment:
@rubik-cube-man This is a slightly complicated case, because we previously used the "stable" channel with `v1.1.2` as the head (it's been a long time since we used this channel name and it was considered unsupported). We wanted to resurrect this channel, however we need to keep the full update graph of the "2.3.x" channel (v2.3.7 is the head) so that it's possible for users to transition from the "2.3.x" -> "stable" channel and upgrade from v2.3.7 -> v2.4.0.

In the community-operators repo the [equivalent PR](https://github.com/k8s-operatorhub/community-operators/pull/4045) has already been merged and is available on OperatorHub. I can confirm that the intended behaviour described above works as expected.

Hopefully the repo maintainers can advise whether we can just ignore this test failure, or if some other course of action is required.

Reply:
Ahhh, I see!

Thanks for explaining what's being done! I'm always curious to learn more! 😄

Reply:
Hello. In this case if you want to put the new bundle in the stable channel, you should point the replaces field to the last bundle in the stable channel prior to this. If you want to allow the upgrades from other channels, you may want to add skip or skiprange fields to the csv as well. If you would just ignore the test, it would keep popping up when you would submit a new bundles.